### PR TITLE
Adding example app walkthrough

### PR DESCRIPTION
**Pull request by CatGarab**

### CODE REVIEW

**Comment by CatGarab** [comment]
[Mission Applications Example in Main Repo](https://trello.com/c/yAoMyPXG/379-mission-applications-example-in-main-repo)

File: docs/app-docs/app-guide.rst [hunk 1]
```diff
@@ -91,12 +91,228 @@ For example::
...
     
-.. todo::
+    use kubos_app::*;
+    use std::fs;
+    
+    struct MyApp;
+    
+    impl AppHandler for MyApp {
+        fn on_boot(&self) {
+            fs::write("home/kubos/test-output", "OnBoot logic\r\n").unwrap();
```
Comment (marshall) [inline docs/app-docs/app-guide.rst]: nit: use absolute paths here in this example (`/home/kubos/test-output`)

File: docs/app-docs/app-guide.rst [hunk 2]
```diff
@@ -91,12 +91,228 @@ For example::
...
-    Here's how we go about coding an actual mission application...
+.. note:: The UUID will be a custom value for each application which is registered
+
+Starting
+~~~~~~~~
+
+We'll go ahead and start our app now to verify it works::
+
+    mutation {
+        startApp(uuid: 60ff7516-a5c4-4fea-bdea-1b163ee9bd7a, runLevel: OnCommand)
```
Comment (marshall) [inline docs/app-docs/app-guide.rst]: nit: quotes around `uuid` value

File: docs/app-docs/app-guide.rst [hunk 3]
```diff
@@ -91,12 +91,228 @@ For example::
...
+Next, we'll transfer the `example-mission-app` file and the `manifest.toml` file into a new
+directory, `/home/kubos/example-app`, on the OBC. 
+
+Registering
+~~~~~~~~~~~
+
+Once both files have been transferred, we can register the application using the ``register`` query::
+
+    mutation {
+        register(path: /home/kubos/example-app) {
```
Comment (marshall) [inline docs/app-docs/app-guide.rst]: nit: quotes around `path` value

File: docs/app-docs/app-guide.rst (hunk 1 above)
Comment (tbrowder1) [reply docs/app-docs/app-guide.rst]: I agree with Marshall

File: docs/app-docs/app-guide.rst (hunk 3 above)
Comment (tbrowder1) [reply docs/app-docs/app-guide.rst]: ditto

File: docs/app-docs/app-guide.rst [hunk 4]
```diff
@@ -91,12 +91,228 @@ For example::
...
+    
+    fn main() {
+        let app = MyApp;
+        app_main!(&app);
+    }
+
+And then update the `config.toml` file to add the `kubos-app` dependency ::
+
+    [dependencies]
+    kubos-app = { path = "../../apis/app-api/rust" }
```
Comment (plauche) [inline docs/app-docs/app-guide.rst]: This should either specify the `kubos-app` dependency using the `git` url or instruct the user to specify the relative path to an already cloned copy of the `kubos` repo.

File: docs/app-docs/app-guide.rst [hunk 5]
```diff
@@ -91,12 +91,228 @@ For example::
...
+        
+And then we'll update the file permissions to allow execution::
+
+    $ chmod +x example-mission-app
+    
+.. note::
 
-    Example Walkthrough
-    //-------------------
+    We're foregoing the usual ".py" extension so that the file name is the same as the Rust example file name.
```
Comment (plauche) [inline docs/app-docs/app-guide.rst]: Is there a specific reason why we have to drop the `.py` extension?

File: docs/app-docs/app-guide.rst [hunk 6]
```diff
@@ -91,12 +91,228 @@ For example::
...
+    
+    def on_boot():
+        
+        file = open("/home/kubos/test-output","w+")
+        file.write("OnBoot logic\r\n")
+        
+    def on_command():
+        
+        file = open("/home/kubos/test-output","w+")
+        file.write("OnBoot logic\r\n")
```
Comment (plauche) [inline docs/app-docs/app-guide.rst]: Should this be `OnCommand logic`?

File: docs/app-docs/app-guide.rst (hunk 5 above)
Comment (CatGarab) [reply docs/app-docs/app-guide.rst]: No, it's just so the name is identical no matter which language you're using for the walkthrough. It has no effect on the actual execution.

**Pull request merged**